### PR TITLE
Updates for downloading and parsing.

### PR DESCRIPTION
**Pull request by lducazu**
All datasources can be downloaded & parsed.
Notable exceptions are DIP and SPIKE. These will be handled separately.

### CODE REVIEW

File: usr/bin/irdata_parse_bar.sh [hunk 1]
```diff
@@ -43,18 +43,11 @@ if [ ! "$DATADIR" ] || [ ! "$FILENAMES" ]; then
 fi
 
 
-#it is assumed that all records in the virushost file refer to protein-protein interactions
-#it is not possible to check for these (as done in the below commented code) because virushost
-#does not follow regular mitab conventions and interactor type is not provided
-#should this change in future, the following code could be uncommented
```
Comment (MaybeJustJames) [inline usr/bin/irdata_parse_bar.sh:49]: Is this no longer true?

File: usr/bin/irdata_parse_quickgo.sh [hunk 2]
```diff
@@ -42,11 +42,7 @@ if [ ! "$DATADIR" ] || [ ! "$FILENAMES" ]; then
     exit 1
 fi
 
-#it is assumed that all records in the reactome file refer to protein-protein interactions
-#it is not possible to check for these (as done in the below commented code) because reactome
-#does not follow regular mitab conventions and interactor type is not provided
-#should this change in future, the following code could be uncommented
```
Comment (MaybeJustJames) [inline usr/bin/irdata_parse_quickgo.sh:48]: Is this no longer true?

File: usr/bin/irdata_parse_uniprot.sh [hunk 3]
```diff
@@ -62,30 +62,31 @@ for FILENAME in $FILENAMES; do
         # Unpack any gzip archives since the slicing of these files is not
         # efficient if done repeatedly.
 
-        if [ "$FILETYPE" = 'dat.gz' ]; then
-            UNPACKED_FILENAME=${FILENAME%.gz}
+        ## Leave file compressed
+        # if [ "$FILETYPE" = 'dat.gz' ]; then
+        #     UNPACKED_FILENAME=${FILENAME%.gz}
```
Comment (MaybeJustJames) [inline usr/bin/irdata_parse_uniprot.sh:67]: Can the commented code in this file be deleted?

File: usr/bin/irdownload [hunk 4]
```diff
@@ -192,16 +192,25 @@ for SOURCE in $SOURCES; do
                         continue
                     fi
 
+                # Download from GCS storage -- requires the Google Gloud SDK
```
Comment (MaybeJustJames) [inline usr/bin/irdownload:195]:
```suggestion
                # Download from GCS storage -- requires the Google Cloud SDK
```

File: usr/bin/irdata-config [hunk 5]
```diff
@@ -212,10 +195,12 @@ CORUM_XML_FILE_PATTERN='*.xml'
 # do we have this file?
 # OK
 DIG_DOWNLOAD_FILES='morbidmap14062010.txt'
-#DIG_DOWNLOAD_FILES='22112010morbidmap.txt'
-DIG_MIRROR='file://'
+# DIG_DOWNLOAD_FILES='22112010morbidmap.txt'
+# DIG_MIRROR='file://'
+# DIG_RELEASE_URL="$DIG_MIRROR/home/irefindex/data/ian/dig/"
+DIG_MIRROR='gs://vib-training-data'
```
[the remark this answers is not in the document]
Comment (MaybeJustJames) [reply usr/bin/irdata-config:201]: Ah I see here: https://github.com/vibbits/irefindex/pull/12/files#diff-f6e08f710350e8b328ab87f7470a6d233e11af0db8118318112d7f10820677eeR196

**Review by MaybeJustJames — CHANGES_REQUESTED** [review]
* I wonder if it makes sense to move `irdata_parse_*.py` into the `irdata` package? Perhaps as an, `irdata.parse` module.
* Why are there shell scripts and Python scripts with the same name? (e.g. `irdata_parse_fly.{sh,py}`)

File: usr/bin/irdata-config (hunk 5 above)
[the remark this answers is not in the document]
Comment (lducazu) [reply usr/bin/irdata-config:201]: The Google Cloud Storage scheme is a temporary solution for storing the static files. In the long run, we should switch to the storage solution offered by UGent.

File: usr/bin/irdata_parse_bar.sh (hunk 1 above)
Comment (lducazu) [reply usr/bin/irdata_parse_bar.sh:49]: This file most likely started as a copy of `irdata_parse_virushost.sh`. There the comment does make sense. For the BAR data source this comment is misleading.

File: usr/bin/irdata_parse_quickgo.sh (hunk 2 above)
Comment (lducazu) [reply usr/bin/irdata_parse_quickgo.sh:48]: The same applies here -- this probably started as a copy of the `irdata_parse_reactome.sh` script.

File: usr/bin/irdata_parse_uniprot.sh (hunk 3 above)
Comment (lducazu) [reply usr/bin/irdata_parse_uniprot.sh:67]: I have a follow up change ready dealing specifically with the more efficient parsing of compressed flat files. The comment will be deleted there.

File: usr/bin/irdownload (hunk 4 above)
Comment (lducazu) [reply usr/bin/irdownload:195]: Ugh...

**Review by MaybeJustJames — APPROVED** [review]

**Pull request merged**